_extras/for_instructors.md

```diff
@@ -0,0 +1,312 @@
+~~~
+apt update
+apt upgrade
+apt install zsh
+sudo -E conda install git
+sudo -E apt-get install tig
+~~~
+
+For the workshop itself, we want to install datalad and its dependencies.
+Run the following lines one by one:
+~~~
+sudo -E apt-get install git-annex
+sudo -E -H pip3 install datalad
+~~~
+
+Depending on the content of your workshop, you might also want to install DataLad extensions or
+other packages. This would be a sensible time to do so. Remember to add `sudo -E` in front of the
+install command in order to make the installation apply to all users.
+
+### 3.5. Set up the default shell
+
+We will set up [`zsh`](https://en.wikipedia.org/wiki/Z_shell) as the default shell for the terminal.
+Currently, when you open a terminal and run `echo $0`, it should print `/usr/bin/bash`, indicating that
+the current default is bash.
+
+In order to change the default, we need to import some `dotfiles` (we will use
+[Pure](https://github.com/sindresorhus/pure)) and tell the terminal and hub how
+to access them in order to set the default shell to `zsh`:
+
+1. Run the following lines one by one in order to import the dotfiles:
+   ~~~
+   sudo mkdir -p "$HOME/.zsh"
+   git clone https://github.com/sindresorhus/pure.git "$HOME/.zsh/pure"
+   ~~~
+2. Create a `.zshrc` file in your HOME directory and copy the content from [this source](https://gist.github.com/mslw/926f1191e61ef2d705fadab66d19b8ba) into the file:
+   ~~~
+   touch "$HOME/.zshrc"
+   nano "$HOME/.zshrc"
+   ~~~
+   This opens the `.zshrc` file for editing in the shell using nano text editor.
+   Now paste the content from the source file in this editor, then hit CTRL+O and Enter (saves edits)
+   and then hit CTRL+X and Enter (exits editor).
+3. Lastly, we also need to make the hub aware of which shell it should use when launching a terminal.
+   We do this with a configuration script. Run the following lines one by one:
+   ~~~
+   touch "$HOME/.jupyter/jupyter_notebook_config.py
+   echo 'c.NotebookApp.terminado_settings = {"shell_command": ["/usr/bin/zsh"]}' > "$HOME/.jupyter/jupyter_notebook_config.py
+   ~~~
+4. This finalizes the shell setup. You can now navigate to the hub's admin panel and restart your server
+   in order for the changes to take effect.
+
+
+### 3.6. Create a global `gitignore`
+
+When we work with `git` repositories and add content or code, we often want certain files
+or directories not to form part of the `git` history. We can achieve this by telling `git`
+to ignore certain files or directories, via the `.gitignore` configuration. Since development
+in JupyterHub often happens via Jupyter Notebooks, which auto-create and -update it's own metadata
+files in the form of `.ipynb_checkpoints` files, we will add those to the global `gitignore` configuration:
+
+~~~
+sudo touch /home/.gitignore_global
+sudo echo ".ipynb_checkpoints" > /home/.gitignore_global
+git config --global core.excludesfile /home/.gitignore_global
```

Comment:
Ah I missed the quotes initially and that's why it didn't work. Then I changed to `/home/.gitignore_global`. I haven't actually opened JH's text editor yet, so let me test it quickly and will comment here with the outcome.

Reply:
I guess it may be sufficient for that file to be readable by normal user, but worth checking.

I also finally found git's [gitignore](https://git-scm.com/docs/gitignore) docs:
> Patterns which a user wants Git to ignore in all situations (e.g., backup or temporary files generated by the user’s editor of choice) generally go into a file specified by core.excludesFile in the user’s ~/.gitconfig. Its default value is $XDG_CONFIG_HOME/git/ignore. If $XDG_CONFIG_HOME is either not set or empty, $HOME/.config/git/ignore is used instead.

So maybe we could get away without setting that config option if we put the ignore file in the default location... but I'm probably optimizing too much now ;)

Reply:
wait, there's something like too much optimization?? I need to learn about that 🙃

Let's use your original suggestion to keep it simple. Then users can also find it easily if they want to amend it.